### ISSUE TITLE
Distinguish a package description from README on absence of the latter

### ISSUE THREAD

**Issue by hyphenrf** [ocaml/ocaml.org#286]
Right now, package overview puts the `description` field text right under `README.md` header. A separation of those two fields seems to make sense, and (perhaps) a way to detect that both fields are the same when a package uses its `README` for the `description` field.

**Comment by tmattio** [comment]
Can you give an example of that? We're not adding the description above the README AFAICT

**Comment by hyphenrf** [comment]
sure, I noticed it first on this package: https://v3.ocaml.org/p/ppx_seq/0.2.0
the `description` field appears under the `README.md` header as it shows

**Comment by hyphenrf** [comment]
Oh, I just checked the source code:

https://github.com/ocaml/v3.ocaml.org-server/blob/a8eb00767d46f983356d287d9b2ce55b21ec684c/src/ocamlorg_web/lib/handler.ml#L423-L427
and it seems descriptions are used as fallback when `README.md` file "doesn't exist".. So I guess the problem is just that. description being under the wrong header on fallback:

https://github.com/ocaml/v3.ocaml.org-server/blob/a8eb00767d46f983356d287d9b2ce55b21ec684c/src/ocamlorg_frontend/pages/package_overview.eml#L18-L20

**Comment by hyphenrf** [comment]
Would you like me to try fixing this myself? ~~I'm thinking of adding a conditional to the template in order to render the correct title depending on whether I'm getting `Some` or `None` (so `Option.map` instead of `Option.value`)~~

**Comment by hyphenrf** [comment]
scratch that, I do need to pass something either way.. so perhaps an extra polyvar/boolean param for title is the right way to approach this. Maybe a title string should be passed along with the discovered filename even. If I do it this way, it could allow you to work out the solution for (#30) with less friction.

What do you think?

**Comment by tmattio** [comment]
I think replacing the description with "This package doesn't have a README" or something similar should be good enough?

**Comment by hyphenrf** [comment]
yeah that sounds alright, I'm onto it

**Comment by hyphenrf** [comment]
.... I don't have enough memory on my laptop to install the deps and build the website. I tried and it eventually crashes midway through and I have to clean up the switch broken package by broken package instead of nuking it and starting fresh.

Anyway, I wrote a patch and unfortunately I can't test it. Someone with a beefier machine can take it for a run and submit the PR 😅 

```patch
diff --git a/src/ocamlorg_frontend/pages/package_overview.eml b/src/ocamlorg_frontend/pages/package_overview.eml
index 7d2fbef..f9ef036 100644
--- a/src/ocamlorg_frontend/pages/package_overview.eml
+++ b/src/ocamlorg_frontend/pages/package_overview.eml
@@ -15,7 +15,7 @@ Package_layout.render
 ~package @@
 <div class="flex lg:space-x-4 xl:space-x-12 flex-col lg:flex-row">
     <div class="flex-1 prose max-w-none">
-        <div class="p-3 bg-body-600 bg-opacity-5 rounded font-semibold mb-8">README.md</div>
+        <div class="p-3 bg-body-600 bg-opacity-5 rounded font-semibold mb-8">README</div>
         <%s! readme %>
     </div>
     <div class="p-3 py-6 lg:p-8 border border-gray-200 text-sm rounded-xl lg:max-w-md w-full">
diff --git a/src/ocamlorg_web/lib/handler.ml b/src/ocamlorg_web/lib/handler.ml
index f9cc404..cf4f44a 100644
--- a/src/ocamlorg_web/lib/handler.ml
+++ b/src/ocamlorg_web/lib/handler.ml
@@ -421,13 +421,10 @@ let package_versioned t kind req =
         | Package -> `Package
         | Universe -> `Universe (Dream.param req "hash")
       in
-      let description =
-        (Ocamlorg_package.info package).Ocamlorg_package.Info.description
-      in
       let* readme =
         let+ readme_opt = Ocamlorg_package.readme_file ~kind package in
         Option.value readme_opt
-          ~default:(description |> Omd.of_string |> Omd.to_html)
+          ~default:("README file not found." |> Omd.of_string |> Omd.to_html)
       in
       let _license = Ocamlorg_package.license_file ~kind package in
       let package_meta = package_meta t package in
```

**Comment by hyphenrf** [comment]
I do think keeping the description when the README file isn't available would be nicer to have, so I also wrote a patch trying that out:

```patch
diff --git a/src/ocamlorg_frontend/ocamlorg_frontend.ml b/src/ocamlorg_frontend/ocamlorg_frontend.ml
index fd88c43..b797cba 100644
--- a/src/ocamlorg_frontend/ocamlorg_frontend.ml
+++ b/src/ocamlorg_frontend/ocamlorg_frontend.ml
@@ -23,9 +23,9 @@ let news_post news = News_post.render news
 let opportunities ?search ?location ~locations opportunities =
   Opportunities.render ?search ?location ~locations opportunities
 
-let package_overview ~readme ~dependencies ~rev_dependencies ~homepages ~source
+let package_overview ~readme ~readme_title ~dependencies ~rev_dependencies ~homepages ~source
     package =
-  Package_overview.render ~readme ~dependencies ~rev_dependencies ~homepages
+  Package_overview.render ~readme ~readme_title ~dependencies ~rev_dependencies ~homepages
     ~source package
 
 let package_documentation ~documentation_status ~toplevel_status ~title ~path
diff --git a/src/ocamlorg_frontend/pages/package_overview.eml b/src/ocamlorg_frontend/pages/package_overview.eml
index 7d2fbef..fad49a5 100644
--- a/src/ocamlorg_frontend/pages/package_overview.eml
+++ b/src/ocamlorg_frontend/pages/package_overview.eml
@@ -2,6 +2,7 @@ let render
 ~documentation_status
 ~toplevel_status
 ~readme
+~readme_title
 ~dependencies
 ~rev_dependencies
 ~homepages
@@ -15,7 +16,7 @@ Package_layout.render
 ~package @@
 <div class="flex lg:space-x-4 xl:space-x-12 flex-col lg:flex-row">
     <div class="flex-1 prose max-w-none">
-        <div class="p-3 bg-body-600 bg-opacity-5 rounded font-semibold mb-8">README.md</div>
+        <div class="p-3 bg-body-600 bg-opacity-5 rounded font-semibold mb-8"><%s! readme_title %></div>
         <%s! readme %>
     </div>
     <div class="p-3 py-6 lg:p-8 border border-gray-200 text-sm rounded-xl lg:max-w-md w-full">
diff --git a/src/ocamlorg_web/lib/handler.ml b/src/ocamlorg_web/lib/handler.ml
index f9cc404..8e89a29 100644
--- a/src/ocamlorg_web/lib/handler.ml
+++ b/src/ocamlorg_web/lib/handler.ml
@@ -424,10 +424,11 @@ let package_versioned t kind req =
       let description =
         (Ocamlorg_package.info package).Ocamlorg_package.Info.description
       in
-      let* readme =
+      let* (readme, readme_title) =
         let+ readme_opt = Ocamlorg_package.readme_file ~kind package in
-        Option.value readme_opt
-          ~default:(description |> Omd.of_string |> Omd.to_html)
+        match readme_opt with
+        | Some doc -> (doc, "README")
+        | None -> (description |> Omd.of_string |> Omd.to_html, "Description")
       in
       let _license = Ocamlorg_package.license_file ~kind package in
       let package_meta = package_meta t package in
@@ -456,8 +457,8 @@ let package_versioned t kind req =
       let* toplevel_status = Ocamlorg_package.toplevel_status ~kind package in
       Dream.html
         (Ocamlorg_frontend.package_overview ~documentation_status
-           ~toplevel_status ~readme ~dependencies ~rev_dependencies ~homepages
-           ~source package_meta)
+           ~toplevel_status ~readme ~readme_title ~dependencies ~rev_dependencies
+           ~homepages ~source package_meta)
 
 let package_doc t kind req =
   let name = Ocamlorg_package.Name.of_string @@ Dream.param req "name" in
```

**Comment by tmattio** [comment]
Fixed in #427. Thanks a lot @hyphenrf and @tatchi!
